src/pyramid_sanity/__init__.py

```diff
@@ -1,44 +1,19 @@
-"""Main integration into Pyramids automatic configuration hoovering."""
-
-from pyramid.config import PHASE3_CONFIG
-from pyramid.interfaces import ITweens
+from pyramid.tweens import MAIN
 
 from pyramid_sanity._egress import EgressTweenFactory
 from pyramid_sanity._ingress import IngressTweenFactory
 from pyramid_sanity._settings import SanitySettings
 
-__all__ = ["includeme"]
-
-
-def _add_tween(config):
-    sanity_settings = config.registry.settings["pyramid_sanity"]
-    tweens = config.registry.queryUtility(ITweens)
-
-    if sanity_settings.ingress_required:
-        tweens.add_explicit("pyramid_sanity.ingress_tween_factory", IngressTweenFactory)
-
-    if sanity_settings.egress_required:
-        for tween_name, tween_factory in tweens.implicit():
-            tweens.add_explicit(tween_name, tween_factory)
-
-        # Question - `warehouse.sanity` adds this after every individual tween,
-        # but I think it only needs to be in once?
-        tweens.add_explicit(
-            "pyramid_sanity.egress_tween_factory", EgressTweenFactory,
-        )
-
 
 def includeme(config):
-    """Configure `pyramid_sanity` to run when called with `config.include`."""
+    settings = SanitySettings.from_pyramid_settings(config.registry.settings)
+
+    # Put the settings into the registry so that other parts of the
+    # pyramid_sanity code can access them.
+    config.registry.settings["pyramid_sanity"] = settings
 
-    config.registry.settings["pyramid_sanity"] = SanitySettings.from_pyramid_settings(
-        config.registry.settings
-    )
+    if settings.ingress_required:
+        config.add_tween("pyramid_sanity.IngressTweenFactory")
 
-    # I'm doing bad things, I'm sorry. - dstufft
-    config.action(
-        ("tween", "pyramid_sanity.tween_factory", True),
-        _add_tween,
-        args=(config,),
-        order=PHASE3_CONFIG,
-    )
+    if settings.egress_required:
+        config.add_tween("pyramid_sanity.EgressTweenFactory", over=MAIN)
```

Comment:
`over=MAIN` means please try to put me last

Reply:
This would also be a useful comment

Reply:
Done